### PR TITLE
FS-2990 & 2989: Devolved authorities & fund specific access 

### PR DESCRIPTION
**Pull request by tferns**
### Change description

- [x] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [ ] Commit messages are meaningful and follow good commit message guidelines

Added `_DEPRECATED_ROLES ` & `_VALID_ROLES ` - we'll log a warning if we come across `_DEPRECATED_ROLES`, as they will need changed to the new groups/roles in Azure AD, but eventually we can probably remove this.  `_VALID_ROLES ` is mostly a 1:1 mapping, except for the deprecated roles, we map the old 3 roles to cof now, as that was the original fund.

Migration script added to change roletype to a string, as it can now support anything - also migrate old 3 roles to cof specific ones, the downgrade should work as intended too, in-case anything goes wrong and we need to revert.

Change `highest_role` to `highest_role_map` - this change was made in tandem with https://github.com/communitiesuk/funding-service-design-utils/pull/85

### Other related PRs:

https://github.com/communitiesuk/funding-service-design-utils/pull/85
https://github.com/communitiesuk/funding-service-design-assessment-store/pull/198
https://github.com/communitiesuk/funding-service-design-assessment/pull/275
https://github.com/communitiesuk/funding-service-design-authenticator/pull/169

### CODE REVIEW

File: core/account.py [hunk 1]
```diff
@@ -86,6 +86,29 @@ def get_bulk_accounts(
         return {"error": "No matching account found"}, 404
 
 
+# old roles, we should remove these once all old roles are removed in azure
+_DEPRECATED_ROLES = {"LEAD_ASSESSOR", "ASSESSOR", "COMMENTER"}
+
+_VALID_ROLES = {
```
Comment (Adam-W1) [inline core/account.py:92]: could we use the az_role_map in config here instead?

Comment (tferns) [reply core/account.py:92]: Ah! yes! good eye

Comment (tferns) [reply core/account.py:92]: Actually, they are slightly different things - `azure_ad_role_map` is a mapping from the azure role to our role. 

This map is a mapping from legacy roles (LEAD_ASSESSOR, ASSESSOR & COMMENTER) to the new roles, but also includes a 1:1 mapping for all other roles, I've reverted this change!

Comment (Adam-W1) [reply core/account.py:92]: okay fair dos, thanks for looking

**Review by Adam-W1 — APPROVED** [review]
LGTM 👍

**Pull request merged**